### DISCUSSION TITLE
How to implement blocks show/hidden by user permission

### DISCUSSION THREAD

**Discussion by emiliosh** [Z3d0X/filament-fabricator#98] [Q&A] [answered]
Hi, I was thinking about how to show/hide blocks by user.  Looking at code, looks like I should rewrite FilamentFabricatorManager::getPageBlocks() and introduce a new method on Blocks something like function hide():bool or show:bool to implement in every Block.   

Have you ever tried something like this?

**Comment by Z3d0X** [comment, marked as answer]
Have you tried using `->hidden(fn ()=> auth()->user()->can('foo'))` on the block (PageBlock getBlockSchema)

**Reply by emiliosh** [reply]
Thanks a lot! it worked!
